File: 백준/Silver/10828. 스택/스택.py
input = __import__('sys').stdin.readline

N = int(input())
stack = []

for i in range(N):
    l = list(input().split())
    if l[0] == 'push':
        stack.append(l[1])
    elif l[0] == 'pop':
        if len(stack) == 0:
            print(-1)
        else:
            print(stack.pop())
    elif l[0] == 'size':
        print(len(stack))
    elif l[0] == 'empty':
        if len(stack) == 0:
            print(1)
        else:
            print(0)
    else:
        if len(stack) == 0:
            print(-1)
        else:
            print(stack[-1])

File: 백준/Silver/10828. 스택/스택.cc
#include <bits/stdc++.h>
using namespace std;

int n, x;
string str;

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);

    stack<int> S;

    cin >> n;
    while(n--) {
        cin >> str;
        if (str == "push") {
            cin >> x;
            S.push(x);
        } else if (str == "pop") {
            if (S.empty()) {
                cout << -1 << '\n';
            } else {
                cout << S.top() << '\n';
                S.pop();
            }
        } else if (str == "size") {
            cout << S.size() << '\n';
        } else if (str == "empty") {
            if (S.empty()) {
                cout << 1 << '\n';
            } else {
                cout << 0 << '\n';
            }
        } else {
            if (S.empty()) {
                cout << -1 << '\n';
            } else {
                cout << S.top() << '\n';
            }
        }
    }
    return 0;
}